runtime/onert/core/include/backend/ITensor.h

```diff
@@ -44,12 +45,19 @@ class ITensor
   virtual bool has_padding() const = 0;
   virtual void access(const std::function<void(ITensor &tensor)> &fn) = 0;
 
-  /**
-   * @brief Return true if the tensor needs dynamic allocation, meaning that during compile-time
-   *        the outpus shape cannot be known and the output shape is calculated during
-   *        kernel execution-time.
-   */
-  virtual bool is_dynamic() const { return false; /* default */ }
+public:
+  // The following methods are needed for dynamic tensor. Dynamic tensor is a tensor whose
+  // the outpus shape cannot be known  during compile-time
+  // and the output shape is calculated during kernel execution-time.
+
+  /// @brief Return true if the tensor needs dynamic allocation
+  virtual bool is_dynamic() const { throw std::runtime_error("not yet implemented"); }
+
+  /// @brief set this tensor dynamic
+  virtual void set_dynamic() { throw std::runtime_error("not yet implemented"); }
+
+  /// @brief set new shape for this tensor
+  virtual void shape(const ir::Shape &) { throw std::runtime_error("not yet implemented"); }
```

Comment:
> @wateret
> Yes, `ir::Shape` of `Graph` starts with high dimension. But ACL's `TensorShape` starts with low dimension. So we should handle it well.
> Additionally, `virtual void dimension(size_t index)` has been used as high dimension.

@ragmani This is a bit late to answer, I think our `ir:Shape` has consistent order, and as `getShape` and `setShape` are virtual, so each backend can reverse the order if it is necessary.

Reply:
@wateret 
>  I think our ir:Shape has consistent order, and as getShape and setShape are virtual, so each backend can reverse the order if it is necessary.

What is consistent order? There is nothing to limit the order. `ir::Shape` can start in any order, either high or low dimension. And it is set different layout.
So `setShape` or `getShape` should have a clear order of whether it starts from high dimension and what layout to follow (for example the order is the same layout of `ITensor`) because it gets or sets `ir::Shape` from `ITensor`.

> This is just about doing two functions with primitive types VS one with a wrapped class.

If you would like to keep primitive types in `ITensor` without "ir::Shape", I do not change it.

Reply:
`IACLTensor::dimension` implementation returns reversed the tensor, since `ir::Shape` and acl_cl's tensor has mismatch in order.

What I want to say is these two ways are basically doing the same thing but in different ways.

Here's pseudocode for that.

```cpp
class ITensor
{
  // virtual void dimension(size_t /* index */, size_t /* dim */)
  // virtual void num_dimensions(size_t /*rank*/)
  virtual void setShape(const ir::Shape &);
};

class CPUTensor
{
  void setShape(const ir::Shape &shape) override;
  {
    _shape = shape;
  }

private:
  ir::Shape _shape;
};

class IACLTensor
{
  void setShape(const ir::Shape &shape) override;
  {
    _shape = createShapeFromACLShape();
  }

  ir::Shape createShapeFromACLInfo()
  {
    // build ir::Shape
    //   - Same way with IACLTensor::dimension() implementation
    //   - Using TensorInfo::num_dimensions() and TensorInfo::dimension()
    //   - need to reverse dimension order
    return created_shape;
  }
};

```

Reply:
> IACLTensor::dimension implementation returns reversed the tensor, since ir::Shape and acl_cl's tensor has mismatch in order.

It means `ITensor::dimension` starts in order high dimension and has the same order layout of `ITensor` what I said. In other words, `setShape()` and `getShape()` using the method has the same order of dimensions as the method now.

Reply:
Yes.